### PR TITLE
Fix main builds

### PR DESCRIPTION
**Pull request by rbtcollins**
Fixes clippy on 1.65, which is used in CI.

### CODE REVIEW

**Review by conradludgate — DISMISSED** [review]

**Comment by rbtcollins** [comment]
@conradludgate hi, with clippy fixed a different error was revealed in rust doc - fixed that too, could I get another approval please?

File: reqwest-tracing/src/reqwest_otel_span_macro.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #[macro_export]
-/// [`reqwest_otel_span!`] creates a new [`tracing::Span`].
+/// [`crate::reqwest_otel_span!`] creates a new [`tracing::Span`].
```
Comment (conradludgate) [inline reqwest-tracing/src/reqwest_otel_span_macro.rs:2]:
```suggestion
/// [`reqwest_otel_span!`](crate::reqwest_otel_span) creates a new [`tracing::Span`].
```
I think is more preferable here

File: reqwest-tracing/src/reqwest_otel_span_macro.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@
 ///
 /// # Macro syntax
 ///
-/// The first argument passed to [`reqwest_otel_span!`] is a reference to an [`reqwest::Request`].
+/// The first argument passed to [`crate::reqwest_otel_span!`] is a reference to an [`reqwest::Request`].
```
Comment (conradludgate) [inline reqwest-tracing/src/reqwest_otel_span_macro.rs:29]: Same here

Comment (rbtcollins) [reply reqwest-tracing/src/reqwest_otel_span_macro.rs]: fixed

**Review by conradludgate — APPROVED** [review]

**Pull request merged**